.github/workflows/scheduled.yml

```diff
@@ -43,22 +41,34 @@ jobs:
 
       - name: Generate matrix
         id: matrix
+        working-directory: ${{ steps.set-up-homebrew.outputs.repository-path }}
         run: |
-          formula="$(find "$(brew --repo "${TAP}")/Formula" -type f | shuf -n "${TEST_COUNT}" | xargs -I{} basename {} .rb)"
-          # shellcheck disable=SC2086
+          # Index all formulae so that we test:
+          #   - formulae 1 to 50 on the first day of the year
+          #   - formulae 51 to 100 on the second day of the year
+          #   - formulae 101 to 150 on the third day of the year
```

Comment:
How about using a modulus here to get a slightly wider spread across the year so we aren't e.g. testing all of `postgresql` in one batch?

Reply:
I understand what you mean about a wider spread, but could you elaborate on using a modulus? (I don't know what that is, but I can think up an implementation otherwise too.)

Reply:
Instead of doing formula 1 to 50: do formula `index % 50`. This will mean that on the first day we get formula 1, 51, 101, second day 2, 52, 102, etc.

Reply:
Does using modulo make it more likely for formula to be missed as new ones are added?

Either way, when formulas are added, some may get skipped through the year? Without using modulo you could overlap by a couple to account for any that are added on a particular day (although you would be testing some formula two days in a row).

Reply:
Ah, sorry, I know `%` as modulo and not modulus. I do use it below to calculate the start index. But `bash` (AFAICT) doesn't make it easy to pick up non-consecutive array elements. Let me think on it a bit.

> Either way, when formulas are added, some may get skipped through the year? 

This is true, but I imagine we'll get to them eventually. We can say something similar about picking 50 randomly anyway.

Reply:
> But `bash` (AFAICT) doesn't make it easy to pick up non-consecutive array elements. I could just construct a `for` loop that does it manually. Let me think on it a bit.

Feels like best options are:
- use Ruby/JS
- ignore me 😁 

either is fine

Reply:
I tried this quickly in `bash`, but it didn't cover all formulae for some reason 😅
```bash
day="$(date +%j)"
formulae_to_test=()

for (( i=0; i < TEST_COUNT; i++ )); do
  index="$(( (day + i * TEST_COUNT) % formulae_count ))"
  formulae_to_test+=("${formulae[${index}]}")
done
```
I may punt on this for now and try changing the spread in a follow-up.

Reply:
Oh, duh. That's because this should probably be
```
day="$(date +%j)"

for (( i=0; i < TEST_COUNT; i++ )); do
  index="$(( (day + i * 365 - 1) % formulae_count ))"
  formulae_to_test+=("${formulae[${index}]}")
done
```